Feature/ot111 57 endpoint creacion slides

wallyjj277 opened the pull request:
creado el endpoint de creacion de slides, con el servicio y el controller, se hicieron algunos fixes al script y a entity slides
enlace de jira: https://alkemy-labs.atlassian.net/browse/OT111-57
evidence:
![slide creada en dbeaver](https://user-images.githubusercontent.com/89556738/146106494-994f82b5-4b72-4c21-9c47-6ad6ca4a2613.png)
![crear slides funciona postman](https://user-images.githubusercontent.com/89556738/146106501-28cf8e3f-520f-4589-93d9-1f1de9568639.png)

wallyjj277 committed:
endpoint crear slides

creacion del Slide controller
creacion del mapper
creacion de dtos
creacion de la clase servicios e implementacion
creacion del endpoint para crear slides
se le aplica swagger siguiendo el ejemplo de category

wallyjj277 committed:
aplico reformat a las clases

aplico reformat a las clases para que queden mas legibles

denisdimarco committed:
OT111-57 - FIX slide.sql, Fix SlideEntity Add Path SecurityConf,

- slide.sql: modificados nombres de columnas con nombres reservados por postgres.
- SlideEntity:  Modificados nombres de atributos, y nombres columnas, agregados offsetDateTime.now()
-SecurityConfiguration. Agregado el path

OsvaldoGWP commented on src/main/java/com/alkemy/ong/model/entity/SlideEntity.java:
```diff
@@ -25,21 +24,22 @@
+
     //Auditory fields
-    private OffsetDateTime created_date;
+    private OffsetDateTime created_date = OffsetDateTime.now();
```
los nombres de los campos de auditoria estan con _ en este archivo  normalizar como lo tenemos en las demas entities

wallyjj277 replied:
resuelto se cambio

OsvaldoGWP commented on src/main/java/com/alkemy/ong/model/mapper/SlideMapper.java:
```diff
@@ -0,0 +1,36 @@
+        return dto;
+    }
+
+    public SlideResponseDTO buildToList(SlideEntity entity) {
```
este no se esta usando por el momento asi q lo saquemos y agregar cuando haga falta si es q hace falta

wallyjj277 replied:
resuelto se elimino se pondra si es requerido en un futuro

OsvaldoGWP commented on src/main/java/com/alkemy/ong/model/request/SlideRequestDTO.java:
```diff
@@ -0,0 +1,20 @@
+    private String text;
+
+    private Integer order;
+    private OrganizationEntity organization;
```
desde el front deberiamos mandar solo el id de la organizacion

wallyjj277 replied:
se cambio por el id de organization

OsvaldoGWP commented on src/main/java/com/alkemy/ong/model/response/SlideResponseDTO.java:
```diff
@@ -0,0 +1,22 @@
+
+    private Integer order;
+
+    private OrganizationEntity organization;
```
en el response recordar de usar dto si tiene relaciones osea aqui deberia usar el dto de organizacion

wallyjj277 replied:
se cambio el entity por el dto

OsvaldoGWP commented on src/main/java/com/alkemy/ong/controller/SlideController.java:
```diff
@@ -0,0 +1,36 @@
+@Tag(name = "slides")
+public class SlideController {
+    private final SlideService slideService;
+
```
Estuve leyendo la tarea  dice q se tiene q hay q mandar una imagen y guardar esa imagen en el s3, porfa chequear eso, lo de imagen lo estuvieron trabajando diego y tomas
NOTA de lo q dice la tarea:
Recibirá la imágen a almacenar en base 64. La misma deberá ser decodeada y almacenada en Amazon S3.

wallyjj277 committed:
resueltos los pedidos de review -1

se cambian los campos de mas que tenian _ y se dejan como estaban por convencion.

se elimina el build to list que no se utilizaba

se manda el id de organizacion en vez del entity completo

se cambia el entity en el response por el dto.

por resolver lo de la imagen base 64, investigando como decodificar y de donde la recibo.

OsvaldoGWP commented on src/main/java/com/alkemy/ong/model/request/SlideRequestDTO.java:
```diff
@@ -0,0 +1,20 @@
+    private String text;
+
+    private Integer order;
+    private long organization_id;
```
aqui respetar la nomenclatura de nombres para la variables seria organizationId tipo Long y agregarle el @NotNull

OsvaldoGWP replied:
despues todo el resto se ve muy bien lo q si falta lo de guardar la imagen del s3 me avisas luego como va eso @wallyjj277 @PadillaTom

wallyjj277 committed:
Create imageUtils.java

creacion de un utils con el codificador de imagenes y el decodificador para que sean reutilizables

wallyjj277 committed:
falta base 64

se hicieron cambios en los dtos

OsvaldoGWP committed:
Se agrega todo

OsvaldoGWP committed:
Se agrega todo
